src/components/Disputes/DisputeTimeline.js

```diff
@@ -1,81 +1,252 @@
 import React from 'react'
-import { GU, textStyle, useTheme } from '@aragon/ui'
-
-import { timeline } from '../../mock-data'
+import {
+  Accordion,
+  GU,
+  textStyle,
+  useTheme,
+  IconFolder,
+  IconFlag,
+  IconGroup,
+  IconVote,
+  IconWrite,
+  Timer,
+} from '@aragon/ui'
+import dayjs from '../../lib/dayjs'
 import Stepper from '../Stepper'
 import Step from '../Step'
+import * as DisputesTypes from '../../types/types'
+import styled from 'styled-components'
+import { useCourtConfig } from '../../providers/CourtConfig'
+import { getDisputeTimeLine } from '../../utils/dispute-utils'
 
-function DisputeTimeline() {
+function DisputeTimeline({ dispute }) {
   const theme = useTheme()
-  const current = 4
+  const roundsLength = dispute.rounds.length
+
+  const courtConfig = useCourtConfig()
+  const disputeTimeLine = getDisputeTimeLine(dispute, courtConfig)
+
+  const reverseTimeLine = disputeTimeLine.reverse().map(item => {
+    if (Array.isArray(item)) {
+      return item.reverse().map(roundPhase => {
+        return roundPhase.reverse()
+      })
+    }
+    return item
+  })
```

Comment:
mmm not sure if i understood

Reply:
`Array#reverse()` doesn’t create a new array (like e.g. what `Array#map()` does), it just modifies the passed one and returns it:

```jsx
const a = [1, 2, 3]
const b = a.reverse()

console.log(b) // [3, 2, 1]
console.log(a) // [3, 2, 1]
console.log(a === b) // true
```

So we need to create a new array if we don’t want to modify in place:

```jsx
const a = [1, 2, 3]
const b = [...a].reverse()

console.log(b) // [3, 2, 1]
console.log(a) // [1, 2, 3]
console.log(a === b) // false
```

Reply:
we are not using the original array for nothing inside the timeline so i think that we are ok  applying the reverse to the original, do you see a potential problem with that?

Reply:
It’s not really a problem right now, it’s more to prevent potential surprises in the future. Here `disputeTimeLine` has been reversed but is not named as such, and someone might try to use it and not expect it to be reversed, leading to subtle bugs. I wish `Array#reverse()` and `Array#sort()` would return a new array instead of editing in place :-(

Reply:
Got it, is cleaner i will do that!